### DISCUSSION TITLE
Whisper

### DISCUSSION THREAD

**Discussion by mpelley** [betalgo/openai#127] [Ideas]
Count this as a vote to add Whisper to the library.

**Comment by kayhantolga** [comment]
Hopefully, coming very soon.

**Reply by mpelley** [reply]
Thank you very much for this interface to Whisper!  It made making an transcription app for my daughter very easy.  She has a speech impediment which causes her speech to be quite slurred.  But Whisper understands her better than most humans do!  So she is dictating her college papers and is very happy.  She says the accuracy is much better than the Google speech-to-text.
Again, thank you very much for making this available to this C# programmer.

**Reply by kayhantolga** [reply]
🙏🏻 it's great to see real-life examples! If you make it open source, I would like to share it as a sample project on the Wiki.

**Comment by kayhantolga** [comment]
Please check the latest version 🎉

**Comment by PaoloStrazzera** [comment]
Great job... greetings from Italy! Could it be possible for you to include a simple example for transcribing an audio file using Whisper? I was not able to find it. Thank you! Ciao, Paolo

**Reply by bozau** [reply]
I got this working for whisper, hopefully this helps you :)


```
string fileName = "c:/users/user/downloads/example.mp3";
FileStream fs = new FileStream( fileName, FileMode.Open, FileAccess.Read );
BinaryReader br = new BinaryReader( fs );
long numBytes = new FileInfo(fileName).Length;
byte[] buff = br.ReadBytes((int)numBytes);

var audioCompletionResult = await openAiService.Audio.CreateTranscription(
    new AudioCreateTranscriptionRequest()
    { 
        Model = Models.WhisperV1, // forced to include this otherwise it throws an exception even if we set default model
        File = buff,
        FileName = "test1.mp3",
        ResponseFormat = "json"
    }
);
```

**Reply by PaoloStrazzera** [reply]
Thank you! I still got two questions, sorry to bother you.
1) where do I set the OpenAI key?
2) why are you setting FileName = "test1.mp3" ? wasn't the binary buffer already passed with File = buff ?

**Reply by PaoloStrazzera** [reply]
I tried to create the following code, based on the example you kindly gave me, but it seems to hang on the call to openAiService.Audio.CreateTranscription() - the passed file is approx 900Kbyte. Am I missing something?

        public async Task<AudioCreateTranscriptionResponse> Execute(string openAI_Key, string mp3FilePath)
        {
            OpenAIService openAiService = new OpenAIService(new OpenAiOptions()
            {
                ApiKey = openAI_Key
            });
            string fileName = mp3FilePath;
            FileStream fs = new FileStream(fileName, FileMode.Open, FileAccess.Read);
            BinaryReader br = new BinaryReader(fs);
            long numBytes = new FileInfo(fileName).Length;
            byte[] buff = br.ReadBytes((int)numBytes);
            AudioCreateTranscriptionRequest audioCreateTranscriptionRequest = new AudioCreateTranscriptionRequest()
            {
                Model = Models.WhisperV1, // forced to include this otherwise it throws an exception even if we set default model
                File = buff,
                FileName = "test1.mp3", // ???
                ResponseFormat = "json"
            };
            var audioCompletionResult = await openAiService.Audio.CreateTranscription(audioCreateTranscriptionRequest);
            return audioCompletionResult;
        }

**Reply by kayhantolga** [reply]
Most of the features have a simple sample in the playground project.
https://github.com/betalgo/openai/blob/master/OpenAI.Playground/TestHelpers/AudioTestHelper.cs

**Comment by Swimburger** [comment]
It would be great if we could upload the audio to Whisper without saving it to disk first.
Can we have overloads that accept a stream? I'd be happy to work on this.

**Reply by Swimburger** [reply]
https://github.com/betalgo/openai/pull/224
